lib/api/bucket_implementation.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:vikunja_app/api/client.dart';
+import 'package:vikunja_app/api/response.dart';
+import 'package:vikunja_app/api/service.dart';
+import 'package:vikunja_app/models/bucket.dart';
+import 'package:vikunja_app/service/services.dart';
+
+class BucketAPIService extends APIService implements BucketService {
+  BucketAPIService(Client client) : super(client);
+
+  @override
+  Future<Bucket> add(int listId, Bucket bucket) {
+    return client
+        .put('/lists/$listId/buckets', body: bucket.toJSON())
+        .then((response) => Bucket.fromJSON(response.body));
+  }
+
+  @override
+  Future delete(int listId, int bucketId) {
+    return client
+        .delete('/lists/$listId/buckets/$bucketId');
+  }
+
+  @override
+  Future<Bucket> get(int listId, int bucketId) {
+    // Might not exist in the API, it isn't in the docs
```

Comment:
That's correct, the endpoint to get a single bucket is not implemented. Might not hurt to have the method in here but it won't work.

Reply:
I shall comment it out then for potential future use.